### PR TITLE
Kronecker generator

### PR DESCRIPTION
**Pull request by rohitvarkey**

### CODE REVIEW

File: src/generators/kronecker.jl [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+"""
+Generates edges for a Kronecker generator graph. Returns an array of 2 rows
+with 1st being the start edge and 2nd row having the end edge.
+"""
+function kronecker(scale::Int64, edgefactor::Int64)
+#Ported from https://gitorious.org/graph500/graph500?p=graph500:graph500.git;a=blob;f=octave/kronecker_generator.m;h=064b6d4f42194cc95ec4ab375dcbee7c3243b233;hb=01a32a9c16994c3cf30bbf4caabf9be128379416.
+    n = 2 ^ scale
+    m = edgefactor * n
+    a,b,c = 0.57, 0.19, 0.19
```
Comment (jpfairbanks) [inline src/generators/kronecker.jl:17]: add as default parameters

File: src/generators/kronecker.jl [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    end
+
+    p = randperm(n)
+    ij = p[ij]
+
+    p = randperm(m)
+    ij = ij[:, p]
+
+    return ij .- 1 #Convert to zero index for Stinger
+end
```
Comment (jpfairbanks) [inline src/generators/kronecker.jl:36]: test for this look at distribution of log degree `hist(log(degree))` should be linear.

**Review by jpfairbanks — COMMENTED** [review]
looks good pending changes

File: src/generators/kronecker.jl (hunk 2 above)
Comment (rohitvarkey) [reply src/generators/kronecker.jl:36]: Should I count both outdegrees and indegrees or just the outdegrees?

**Comment by jpfairbanks** [comment]
They should agree.

**Pull request merged**